pkg/sass-parser/lib/src/statement/variable-declaration.ts

```diff
@@ -0,0 +1,228 @@
+        : '') +
+      '$' +
+      (this.raws.variableName?.value === this.variableName
+        ? this.raws.variableName.raw
+        : sassInternal.toCssIdentifier(this.variableName))
+    );
+  }
+  set prop(value: string) {
+    throw new Error("VariableDeclaration.prop can't be overwritten.");
+  }
+
+  get value(): string {
+    return this.expression.toString();
+  }
+  set value(value: string) {
+    this.expression = {text: value};
+  }
+
+  get important(): boolean {
+    // TODO: Return whether `this.expression` is a nested series of unbracketed
+    // list expressions that ends in the unquoted string `!important` (or an
+    // unquoted string ending in " !important", which can occur if `value` is
+    // set // manually).
+    throw new Error('Not yet implemented');
+  }
+  set important(value: boolean) {
+    // TODO: If value !== this.important, either set this to a space-separated
+    // list whose second value is `!important` or remove the existing
+    // `!important` from wherever it's defined. Or if that's too complex, just
+    // bake this to a string expression and edit that.
+    throw new Error('Not yet implemented');
+  }
+
+  get variable(): boolean {
+    return true;
+  }
+
+  constructor(defaults: VariableDeclarationProps);
+  /** @hidden */
+  constructor(_: undefined, inner: sassInternal.VariableDeclaration);
+  constructor(
+    defaults?: VariableDeclarationProps,
+    inner?: sassInternal.VariableDeclaration
+  ) {
+    super(defaults as unknown as postcss.DeclarationProps);
+    this.raws ??= {};
+
+    if (inner) {
+      this.source = new LazySource(inner);
+      this.namespace = inner.namespace ? inner.namespace : undefined;
+      this.variableName = inner.name;
+      this.expression = convertExpression(inner.expression);
+      this.guarded = inner.isGuarded;
+      this.global = inner.isGlobal;
+    } else {
+      this.guarded ??= false;
+      this.global ??= false;
+    }
+  }
+
+  clone(overrides?: Partial<VariableDeclarationProps>): this {
+    return utils.cloneNode(
+      this,
+      overrides,
+      [
+        'raws',
+        {name: 'namespace', explicitUndefined: true},
+        'variableName',
+        'expression',
+        'guarded',
+        'global',
+      ],
+      ['value']
+    );
+  }
+
+  toJSON(): object;
+  /** @hidden */
+  toJSON(_: string, inputs: Map<postcss.Input, number>): object;
+  toJSON(_?: string, inputs?: Map<postcss.Input, number>): object {
+    return utils.toJSON(
+      this,
+      ['namespace', 'variableName', 'expression', 'guarded', 'global'],
+      inputs
+    );
+  }
+
+  /** @hidden */
+  toString(): string {
+    return (
+      this.prop +
+      (this.raws.between ?? ': ') +
+      this.expression +
+      (this.raws.flags?.value?.guarded === this.guarded &&
+      this.raws.flags?.value?.global === this.global
+        ? this.raws.flags.raw
+        : (this.guarded ? ' !default' : '') + (this.global ? ' !global' : '')) +
```

Comment:
Should there be documentation on the order in which these are rendered in the case where both are true but raws are empty or mismatch?

Reply:
Nah, I don't think documenting the default rendering is important when it's easy to control if you want to.